### ISSUE TITLE
Actions function signature

### ISSUE THREAD

**Issue by helfi92** [edriang/react-connect-context-hooks#9]
I'm currently migrating an app from redux/redux-thunk to react-connect-context-hooks.

With redux-thunk actions have the following format:

```jsx
const action = () => (dispatch, getState) => {};
```

In order to migrate to react-connect-context-hooks, a user needs to switch it to 

```jsx
const action = (dispatch, getState) => () => {};
```

Making this change for a few actions is fine but for 500+ actions it becomes inconvenient for users migrating to this library. Are you open to switching the order so that users who are migrating to react-connect-context-hooks don't feel like it's a blocker? Thanks!

**Comment by adrian-marcelo-gallardo** [comment]
Hey @helfi92, thanks for asking.

I see this might result painful for your use-case, but I think changing this in the library would be a major breaking change.

When I created the library, I didn't do it as a port (or as a direct alternative) for `redux`, and therefore I built it guiding myself based on previous experience and "common sense".

In the case of the `thunk`, it is still hard to me to reason about it when I see the functional notation, since the application-domain parameters comes first and you don't know where the `dispatch` and `getState` methods comes from; e.g. `const action = (param1, param2) => (dispatch, getState) => {}`.

This works in redux because you are required to invoke the action within the `dispatch` method, e.g.: `dispatch(action(param1, param2))`, but I liked the idea to absolutely forget about the dispatcher in your application, and allow the developer to simply call `action(param1, param2)`. This is why the library implementation is just based on Closures, which (at least from my POV) is easier to understand without knowing about the `dispatch` internals.
